### PR TITLE
Increase PyYAML limit as awscli

### LINKED ISSUE #441: eb cli pip install not working
### Description
Starting from today when installing eb cli using `pip install --upgrade pip awsebcli` on the python:3 container it fails with:

```
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [68 lines of output]
      /tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/config/setupcfg.py:293: _DeprecatedConfig: Deprecated config in `setup.cfg`
      !!
      
              ********************************************************************************
              The license_file parameter is deprecated, use license_files instead.
      
              By 2023-Oct-30, you need to update your project and remove deprecated calls
              or your builds will no longer be supported.
      
              See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
              ********************************************************************************
      
      !!
        parsed = self.parsers.get(option_name, lambda x: x)(value)
      running egg_info
      writing lib3/PyYAML.egg-info/PKG-INFO
      writing dependency_links to lib3/PyYAML.egg-info/dependency_links.txt
      writing top-level names to lib3/PyYAML.egg-info/top_level.txt
      Traceback (most recent call last):
        File "/usr/local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/usr/local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/local/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 338, in run_setup
          exec(code, locals())
        File "<string>", line 271, in <module>
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/__init__.py", line 107, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
                 ^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 1234, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 314, in run
          self.find_sources()
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 322, in find_sources
          mm.run()
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 551, in run
          self.add_defaults()
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 589, in add_defaults
          sdist.add_defaults(self)
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/command/sdist.py", line 104, in add_defaults
          super().add_defaults()
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/sdist.py", line 251, in add_defaults
          self._add_defaults_ext()
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/sdist.py", line 336, in _add_defaults_ext
          self.filelist.extend(build_ext.get_source_files())
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<string>", line 201, in get_source_files
        File "/tmp/pip-build-env-bw9t4kgw/overlay/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 107, in __getattr__
          raise AttributeError(attr)
      AttributeError: cython_sources
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```
### Steps to reproduce

```
docker run -it --rm python:3 bash
pip install --upgrade pip awsebcli
```
### Additional environment details

1. OS: Linux (docker container)

### PR DESCRIPTION
**Pull request by mbonfoster**
Should fix #441

*Description of changes:*
This PR updates PyYAML limit in order to match with awscli.
https://github.com/aws/aws-cli/pull/8037

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ahmaze** [comment]
This is the related issue in pyyaml repo: https://github.com/yaml/pyyaml/issues/724

**Comment by mbonfoster** [comment]
@Mickeypeng any chance of reviewing/merging this PR?

**Comment by justinpincar** [comment]
Thank you for the fix! This issue is preventing us from deploying; would love to see this merged soon~

**Comment by Mickeypeng** [comment]
@NihalM99

File: requirements.txt [hunk 1]
```diff
@@ -9,5 +9,5 @@ semantic_version == 2.8.5
 six>=1.11.0,<1.15.0
```
Comment (chenrui333) [inline requirements.txt:9]: can we maybe also bump `six` constraint as well?

seeing some issue in https://github.com/Homebrew/homebrew-core/pull/136977

```
pkg_resources.DistributionNotFound: The 'six<1.15.0,>=1.11.0' distribution was not found and is required by awsebcli
```

Comment (chenrui333) [reply requirements.txt:9]: happy to file a different PR on it.

Comment (mbonfoster) [reply requirements.txt:9]: @chenrui333 I think that maybe another PR would be better

Comment (chenrui333) [reply requirements.txt:9]: Totally, I will do a different PR then.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
actually found a dependency conflict with docker-compose v1

relates to https://github.com/docker/compose/issues/9114

**Comment by mbonfoster** [comment]
@Mickeypeng @NihalM99 updates? I think that merging this PR will help many people that's having issues with deployment. Of course it's also necessary to check the conflict with docker-compose that @chenrui333 is mentioning, but merging this PR will help on many other cases.

**Review by NihalM99 — APPROVED** [review]

**Comment by gabrielmontenegro-eye** [comment]
Thank you on this effort guys. Just reinforcing that this issue is also preventing us from deploying an important hotfix.

**Comment by Mickeypeng** [comment]
Seems like the test failed to run because it was from branch in another repo. Let me pull a new PR for testing, and then I will merge this one.

**Comment by Mickeypeng** [comment]
https://github.com/aws/aws-elastic-beanstalk-cli/pull/449 Same changes passed tests. Merging now

**Comment by gabrielmontenegro-eye** [comment]
Yey! Thank you @Mickeypeng and @NihalM99 . How long will it take for a new tag / package version?

**Pull request merged**